ibm/data_source_ibm_tg_gateways_test.go

```diff
@@ -0,0 +1,34 @@
+package ibm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccIBMTransitGatewaysDataSource_basic(t *testing.T) {
+	resName := "data.ibm_tg_gateways.test1"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckIBMTransitGatewaysDataSourceConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(resName, "transit_gateways.0.name"),
+					resource.TestCheckResourceAttrSet(resName, "transit_gateways.0.location"),
+					resource.TestCheckResourceAttrSet(resName, "transit_gateways.0.global"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIBMTransitGatewaysDataSourceConfig() string {
+	// status filter defaults to empty
```

Comment:
Does this test case works if you don't have any gateways in our account I think  it fails at this condition checks
resource.TestCheckResourceAttrSet(resName, "transit_gateways.0.name"),

Change the testcase to create and then fetch the gateways some reference
https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/ibm/data_source_ibm_is_subnets_test.go

Reply:
Added code to create and check gateways list. Tested with no transit gateways in account and test case is working fine.